### PR TITLE
Several fixes and improvements.

### PR DESCRIPTION
**Pull request by s0undt3ch**
Several fixes and improvements.

### CODE REVIEW

File: saltcloud/clouds/gogrid.py [hunk 1]
```diff
@@ -108,6 +108,13 @@ def create(vm_):
     '''
     Create a single VM from a data dict
     '''
+    deploy = config.get_config_value('deploy', vm_, __opts__)
+    if deploy is True and salt.utils.which('sshpass') is None:
+        raise SaltCloudSystemExit(
+            'Cannot deploy salt in a VM if the \'sshpass\' binary is not '
```
Comment (techhat) [inline saltcloud/clouds/gogrid.py:114]: I like that you specify it to be a binary.

Comment (s0undt3ch) [inline saltcloud/clouds/gogrid.py:114]: Well, it was a bit hard to find out why the VM's weren't spinning ;)

RTFM always helps... :smile:

Comment (aleksandarnenov) [inline saltcloud/clouds/gogrid.py:114]: Thanks!

Best Regards,

On Mon, Apr 15, 2013 at 3:53 AM, Pedro Algarvio notifications@github.comwrote:

> In saltcloud/clouds/gogrid.py:
> 
> > @@ -108,6 +108,13 @@ def create(vm_):
> >      '''
> >      Create a single VM from a data dict
> >      '''
> > -    deploy = config.get_config_value('deploy', vm_, **opts**)
> > -    if deploy is True and salt.utils.which('sshpass') is None:
> > -        raise SaltCloudSystemExit(
> > -            'Cannot deploy salt in a VM if the \'sshpass\' binary is not '
> 
> Well, it was a bit hard to find out why the VM's weren't spinning ;)
> 
> RTFM always helps... [image: :smile:]
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/saltstack/salt-cloud/pull/533/files#r3787574
> .

## 

_Aleksandar N._

IT Consultant, Cloud Computing & Linux Expert

_E-mail: alex@cloudwebops.com_\* *

_LinkedIN: *_http://www.linkedin.com/in/cloudwebops*http://www.linkedin.com/in/cloudwebops

_Twitter: *_https://twitter.com/cloudwebops*https://twitter.com/cloudwebops

## 

This e-mail is confidential and may also contain privileged information. If
you are not the intended recipient, you are not authorized to read, print,
save, process or disclose this message. If you have received this message
by mistake, please inform the sender immediately and delete this e-mail,
its attachments and any copies. Thank you!

Comment (s0undt3ch) [inline saltcloud/clouds/gogrid.py:114]: @cloudwebops Anytime!

**Pull request merged**